test/Feature/UpsertSpec.hs

```diff
@@ -44,6 +44,12 @@ spec =
             , matchHeaders = ["Preference-Applied" <:> "resolution=merge-duplicates", matchContentTypeJson]
             }
 
+        it "succeeds when the payload has no elements" $
+          request methodPost "/articles" [("Prefer", "return=representation"), ("Prefer", "resolution=merge-duplicates")]
+            [json|[]|] `shouldRespondWith`
+            [json|[]|] { matchStatus = 201 , matchHeaders = [matchContentTypeJson] }
```

Comment:
HTTP 201 Created is a bit misleading, an HTTP 200 OK would be better. You could conditionally change the status by comparing if `queryTotal = 0` I believe.

Reply:
I agree with you that HTTP 201 is misleading.

I've tried to not to de-rail, from current behavior and I had found this test:
```
-- test/Feature/InsertSpec.hs (lines 131-137)
it "returns empty array when no items inserted, and return=rep" $ do
  p <- request methodPost "/no_pk"
               [("Prefer", "return=representation")]
               [json| [] |]
  liftIO $ do
    simpleBody p `shouldBe` [json| [] |]
    simpleStatus p `shouldBe` created201
```

Which I would argue is similar to the one created in this PR. 
Let me know, which option do you think it's more fitting: Change both behaviors to 200, change just this PR to 200 or keep it as is for consistency.

Reply:
Both behaviors to 200 seems correct. My reasoning is that with [201](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) we can include a `Location` header for identifying the resource, and with `[]` as the response we can't do that.